scipy/stats/tests/test_fit.py

```diff
@@ -109,6 +117,16 @@ def test_cont_fit(distname, arg, method):
                     " test nevertheless.]")
             pytest.xfail(msg)
 
+    if distname in xslow_fits[method]:
+        # Skip xslow fits unless SCIPY_XLSOW=1
+        try:
+            xslow = not int(os.environ['SCIPY_XSLOW'])
```

Comment:
Considering the use of `not`, calling this `xslow` (instead of say, `skip_xslow`) here seems a bit backwards, but I'm just following the template above. I can change both if preferred.

Reply:
On top of that, I wonder if we could also avoid catching a generic `Exception`, which is always nice to avoid, with something like:

```python
skip_xslow = not os.getenv(['SCIPY_XSLOW'], default=False)
```

Still annoying for the brain to parse though.

Apart from that, the current version of the diff looks much closer to what Ralf was asking for, IMO. I suppose I might have a moderate concern that this will effectively double the number of PRs that just add slight tweaks to i.e., test tols by having a need to bump fail times or recategorize tests, but perhaps worth a try as you note.

Reply:
Yeah, I didn't know `getenv` had a default like `getattr`; I should have looked. I'll do that.

> but perhaps worth a try as you note.

Yeah, let's see. My hope is that this will just make PR authors more conscious and fix things like this before they are merged.

Reply:
OK, I think the logic is much cleaner now.